aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -170,11 +170,14 @@ void log_sigmoid_backward_kernel(TensorIterator& iter) {
 // prelu forward
 // -----------------------------------
 void launch_prelu_cuda_kernel_share_weights(TensorIteratorBase &iter, const TensorBase &weight) {
-  AT_DISPATCH_FLOATING_TYPES_AND(at::ScalarType::Half, iter.input_dtype(), "prelu_cuda", [&] {
-    const auto *weight_data = weight.data_ptr<scalar_t>();
+  AT_DISPATCH_FLOATING_TYPES_AND(at::ScalarType::Half,
+                                 iter.input_dtype(), "prelu_cuda", [&] {
+    using opmath_t = at::opmath_type<scalar_t>;
+    const auto *weight_data = weight.data_ptr<opmath_t>();
```

Comment:
you can't just change pointer dtype (this is what's causing your errors). You should roll back this change, and then the errors will disappear. Also, technically, you don't need `opmath` here at all, because there's a single computation operation that happens (`weight*input`), there are no intermediates.

Reply:
Hi, @ngimel! Got that. I've updated it. Thanks, for sharing :)